addons/account/demo/account_demo.py

```diff
@@ -61,9 +61,13 @@ def _post_load_demo_data(self, company=False):
     def _get_demo_data_products(self, company=False):
         prod_templates = self.env['product.product'].search(self.env['product.product']._check_company_domain(company))
         if self.env.company.account_sale_tax_id:
-            prod_templates.write({'taxes_id': [Command.link(self.env.company.account_sale_tax_id.id)]})
+            prod_templates_sale = prod_templates.filtered(
+                lambda p: not p.taxes_id.filtered(lambda t: t.company_id==self.env.company))
```

Comment:
I don't understand the filtering on the taxes.
What are you trying to fix, the message is really not clear.

Reply:
If I am adding a product with tax in demo/data like in https://github.com/odoo/odoo/pull/139514/files#diff-364ad0964b5350aced016829441517ae29f74f01838d898ba45887dc94781ac0R48, its tax would not be overwritten.  I see also that there is another problem with this, because it won't be applied on already installed CoAs.